### PR TITLE
ESP32: Update CHIP code to use latest ESP-IDF v4.2 release

### LINKED ISSUE #4811: ESP32: Update CHIP code to use ESP-IDF v4.2
#### Problem
CHIP uses ESP-IDF v4.1 and v4.2 is available

 #### Proposed Solution
Update CHIP code to use ESP-IDF v4.2 instead of v4.1

### PR DESCRIPTION
**Pull request by dhrishi**
#### Problem
Need to update CHIP to use ESP-IDF v4.2

Tests done:

- Tested compilation and functionality of all-clusters-app with ESP-IDF release/v4.2 branch

 Fixes #4811

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Comment by dhrishi** [comment]
There are no changes required in the CHIP code for this ESP-IDF version update

**Comment by dhrishi** [comment]
@rwalker-apple Do we need to do update the IDF version manually on some server? I remeber something being done when upgrading from v4.0 to v4.1, though not sure :-)

File: integrations/docker/images/chip-build-esp32/Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ RUN set -x \
     && DEBIAN_FRONTEND=noninteractive apt-get install -y python \
     && mkdir -p /opt/espressif \
     && cd /opt/espressif \
-    && git clone --progress -b release/v4.1 https://github.com/espressif/esp-idf.git \
+    && git clone --progress -b release/v4.2 https://github.com/espressif/esp-idf.git \
```
Comment (andy31415) [inline integrations/docker/images/chip-build-esp32/Dockerfile:10]: you need to update dockerfile version for this.

Comment (andy31415) [reply integrations/docker/images/chip-build-esp32/Dockerfile:10]: And then a followup PR will be required to update all our CI and vscode to the new version.

Logic being that we want to test compilation and docker CI images will also be used to generate release artifacts.

Comment (dhrishi) [reply integrations/docker/images/chip-build-esp32/Dockerfile:10]: @andy31415  I have incremented the dockerfile version

**Review by mspang — APPROVED** [review]

File: integrations/docker/images/chip-build/version [hunk 2]
```diff
@@ -1 +1 @@
-0.4.19
+0.4.20
```
Comment (bzbarsky-apple) [inline integrations/docker/images/chip-build/version:1]: Don't we need to at least bump `integrations/docker/images/chip-build-esp32/version` as well?  If not all the versions under `integrations/docker/images/`.

Comment (dhrishi) [reply integrations/docker/images/chip-build/version:1]: Hi @bzbarsky-apple  Looks like it is a symlink. So, no changes required?

```
$ ls -l integrations/docker/images/chip-build-esp32/version
lrwxrwxrwx 1 hrishi hrishi 21 Dec 11 00:12 integrations/docker/images/chip-build-esp32/version -> ../chip-build/version
```

Comment (bzbarsky-apple) [reply integrations/docker/images/chip-build/version:1]: Ah, excellent.

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**